doc/ACL_pytest.md

```diff
@@ -0,0 +1,326 @@
+
+#### Test structure
+
+```
+ansible/
+    roles/
+        test/
+            tasks/
+                acltb.yml
+                acl/
+                    acltb/
+                        acltb_config.yml
+                        acltb_config_generate.yml
+                        acltb_run_test.yml
+                        acltb_test_basic.yml
+                        acltb_test_port_toggle.yml
+                        acltb_test_reboot.yml
+                   acltb_expect_messages.txt
+                   acltb_ignore_messages.txt
+                   acltb_match_messages.txt
+                   acltb_test_rules.json
+                   acltb_test_rules_part_1.json
+                   acltb_test_rules_part_2.json
+            files/
+                acstests/
+                    acltb_test.py
+```
+
+- *acltb.yml*
+
+    The main entry point for the test.
+
+- *acltb_config.yml*/*acltb_config_generate.yml*
+
+    Setting up the test variables; ACL tables, rules configuration generation
+
+- *acltb_run_test.yml*
+
+    Apply generated configuration & run *acltb_test.basic.yml*, *acltb_test_port_toggle.yml*, *acltb_test_reboot.yml*
+
+- *acltb_test_<basic|port_togle|reboot.yml*
+
+    Actual tests
+
+### Scope of this document
+
+- we will cover everything that acltb.yml roles does and suggest an approach to convert it to pytest
+- we will skip basic sanity checks required for all tests - this is not the scope of this document
+
+### Requirements
+
+- Provide the same coverage
+- Test cases isolation, e.g. ability to run single traffic test cases without a need to run whole test
+- Avoid duplications like in acltb_test_*.yml playbooks
+
+
+### Framework requirements
+
+pytest-ansible plugin does not provide all the functionality it may be required for implementing ACL test
+
+##### 'ansible_*' fixtures scope limitation
+
+Because of 'ansible_adhoc' fixture is scoped to a function it limits the possibility to use this fixture in class/module/session scoped setup/teardown style fixtures.
+
+There is no obious reason why it was limited to 'function'. Unless SONiC system test framework uses 'ansible_adhoc' in a way it breaks the 'session' scope, we could override the fixture with 'session' scope in 'conftest.py'.
+
+NOTE: *We will skip this part here in the design right now until we reach some decision on this limitation; Below fixutres in this design depend on 'dut' host provided by 'ansible_adhoc'*
+
+
+##### Common tasks missing
+
+The following has to be implemented in the same way as in relevant playbook roles:
+
+- reboot
+- reload config
+- port toggle
+
+### Design
+
+#### Directory structure
+
+The test will be structured as follows:
+
+```
+tests/
+    acstests/ -> ../ansible/roles/test/files/acstests/
+    common/
+        __init__.py
+        reload_config.py
+        reboot.py
+        port_toggle.py
+    acl/
+        conftest.py
+        test_acl.py
+        templates/
+        loganalyzer/
```

Comment:
Log analyzer is not acl specific. It is better to put it under `common`.

Reply:
Sorry, the design misses the description for *loganalyzer* directory, This directory is for loganalyzer match/expect/ignore files. I will add that